agent/src/main/java/com/teamscale/jacoco/agent/options/AgentOptions.java

```diff
@@ -308,17 +323,33 @@ private File getTempFile(final String prefix, final String extension) {
 	 * Returns in instance of the agent that was configured. Either an agent with interval based line-coverage dump or
 	 * the HTTP server is used.
 	 */
-	public AgentBase createAgent(
-			Instrumentation instrumentation) throws UploaderException, CoverageGenerationException {
+	public AgentBase createAgent(Instrumentation instrumentation) throws UploaderException {
 		if (useTestwiseCoverageMode()) {
-			return new TestwiseCoverageAgent(this, new TestExecutionWriter(getTempFile("test-execution", "json")));
+			JaCoCoTestwiseReportGenerator reportGenerator = new JaCoCoTestwiseReportGenerator(
```

Comment:
I changed this so the report generator no longer fills the cache on construction, which was unexpected to me. Instead it fills it the first time you try to use it. Since the exec file strategy doesn't use it, it never does :-)